### ISSUE TITLE
The Kullback-Leibler divergence

### ISSUE THREAD

**Issue by darkpromise98** [MLforHealth/S2SD#3]
I found the Kullback-Leibler divergence in this paper is different from the standard form.  There is one more operation (a log-operation on A). Why do it this way? 
https://github.com/MLforHealth/S2SD/blob/1ef26e4e273019575f3b5dfbb789fa4952eb9646/criteria/s2sd.py#L133-L135

What's more, in previous KD methods, logits or features of teacher (target) network is in the first position of kl_div function, and student (source) is in the second. But in this paper, the order is reverse. Why?
https://github.com/MLforHealth/S2SD/blob/1ef26e4e273019575f3b5dfbb789fa4952eb9646/criteria/s2sd.py#L111

**Comment by Confusezius** [comment]
Hi there!

The PyTorch  KLDiv performs the normal KLD, but needs the first input to be a log (see the docs). So in the end, we perform a standard KLD. Regarding the ordering, matching is performed either way, and we found both settings to boost performance notably anyway. The current setting worked experimentally slightly better, treating the target similarities as KLD targets :).

**Comment by darkpromise98** [comment]
Thanks a lot.
